cocotb/handle.py

```diff
@@ -74,18 +74,17 @@ def __init__(self, handle, path):
             path (str): Path to this handle, ``None`` if root.
         """
         self._handle = handle
-        self._len = None
-        self._sub_handles = {}  # Dictionary of children
-        self._invalid_sub_handles = set()  # Set of invalid queries
-
-        self._name = simulator.get_name_string(self._handle)
-        self._type = simulator.get_type_string(self._handle)
-        self._fullname = self._name + "(%s)" % self._type
-        self._path = self._name if path is None else path
-        self._log = SimLog("cocotb.%s" % self._name)
+        self._len : int = None  #: The "length" (the number of elements) of the underlying object. For vectors this is the number of bits.
+        self._sub_handles : dict = {}  #: Dictionary of this handle's children.
+        self._invalid_sub_handles : set = set()  #: Python :keyword:`set` of invalid queries, for caching purposes.
```

Comment:
These two members strike me as actually private, since they're just a cache.

Reply:
We do have other docstrings (``_discover_all()``) pointing to ``_sub_handles``, so ... I'm undecided.
I agree that ``_invalid_sub_handles`` isn't something a user is going to be interested in.